test/e2e/curated_packages_test.go

```diff
@@ -4,37 +4,174 @@
 package e2e
 
 import (
+	"context"
 	"fmt"
+	"strconv"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/aws/eks-anywhere/internal/pkg/api"
 	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/test/framework"
 )
 
 const (
-	eksAnywherePackagesHelmChartName = "eks-anywhere-packages"
-	eksAnywherePackagesHelmUri       = "oci://public.ecr.aws/l0g8r8j6/eks-anywhere-packages"
-	eksAnywherePackagesHelmVersion   = "0.1.14-eks-a-v0.0.0-dev-build.3481"
-	eksAnywherePackagesBundleUri     = "oci://public.ecr.aws/l0g8r8j6/eks-anywhere-packages-bundles:v1-21-latest"
-
-	eksaPackageControllerHelmChartName = "eks-anywhere-packages"
-	eksaPackageControllerHelmURI       = "oci://public.ecr.aws/eks-anywhere/eks-anywhere-packages"
-	eksaPackageControllerHelmVersion   = "0.1.10-eks-a-10"
-	eksaPackageBundleBaseURI           = "oci://public.ecr.aws/eks-anywhere/eks-anywhere-packages-bundles"
+	EksaPackagesNamespace              = "eksa-packages"
+	EksaPackageControllerHelmChartName = "eks-anywhere-packages"
+	EksaPackagesSourceRegistry         = "public.ecr.aws/l0g8r8j6"
+	EksaPackageControllerHelmURI       = "oci://" + EksaPackagesSourceRegistry + "/eks-anywhere-packages"
+	EksaPackageControllerHelmVersion   = "0.1.14-eks-a-v0.0.0-dev-build.3624"
+	EksaPackageBundleURI               = "oci://" + EksaPackagesSourceRegistry + "/eks-anywhere-packages-bundles"
 )
 
-var (
-	eksaPackageControllerHelmValues = []string{}
-	eksAnywherePackagesHelmValues   = []string{"sourceRegistry=public.ecr.aws/l0g8r8j6"}
-)
+var EksaPackageControllerHelmValues = []string{"sourceRegistry=public.ecr.aws/l0g8r8j6"}
+
+type resourcePredicate func(string, error) bool
```

Comment:
Need to export this if its used by the public API (`WaitForResource` consumes it).